django/utils/html.py

```diff
@@ -324,6 +325,7 @@ def __call__(self, text, trim_url_limit=None, nofollow=False, autoescape=False):
             ]
         )
 
+    @functools.cache
     def handle_word(
```

Comment:
> [Because it never needs to evict old values](https://docs.python.org/3/library/functools.html#functools.cache)

Wouldn't this unbounded cache cause memory exhaustion problems for long running Python processes that use `urlize` on user controlled data?

It seems that we use a single `Urlizer` instance relied on by `urlizer` which we keep a reference to at the module level so the cache entries in `handle_word` would never be released